File: 2405.py
class Solution:
    def partitionString(self, s: str) -> int:
        sum = 1
        uniques = set()
        for c in s:
            if c in uniques:
                uniques.clear()
                uniques.add(c)
                sum += 1
            elif c not in uniques:
                uniques.add(c)
        return sum


assert Solution().partitionString("abacaba") == 4
assert Solution().partitionString("ssssss") == 6

File: 2405.cpp
#include <set>
#include <string>
using namespace std;
class Solution {
public:
  int partitionString(string s) {
    int uniques = 0;
    int ans = 1;
    for (char c : s) {
      if (uniques & (1 << (c - 'a'))) {
        uniques = 0;
        ans += 1;
      }
      uniques += (1 << (c - 'a'));
    }
    return ans;
  }
};